cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -522,20 +550,64 @@ exports.certificateRequestHandler = async function (event, context) {
   var physicalResourceId;
   var certificateArn;
   const props = event.ResourceProperties;
+  const [app, env, domain] = [props.AppName, props.EnvName, props.DomainName];
+  domainTypes = {
+    EnvDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${env}.${app}.${domain}`),
+      domain: `${env}.${app}.${domain}`,
+    },
+    AppDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${app}.${domain}`),
+      domain: `${app}.${domain}`,
+    },
+    RootDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${domain}`),
+      domain: `${domain}`,
+    },
+    OtherDomainZone: { regex: new RegExp(`.*`) },
+  };
 
   try {
+    var certDomain = `${props.EnvName}.${props.AppName}.${props.DomainName}`;
+    var aliases = await getAllAliases(props.Aliases);
     switch (event.RequestType) {
       case "Create":
+        certificateArn = await requestCertificate(
+          event.RequestId,
+          props.AppName,
+          props.EnvName,
+          certDomain,
+          [...aliases],
+          props.EnvHostedZoneId,
+          props.RootDNSRole,
+          props.Region
+        );
+        responseData.Arn = physicalResourceId = certificateArn;
+        break;
       case "Update":
+        // Exit early if cert doesn't change.
+        if (event.OldResourceProperties) {
+          var prevAliases = await getAllAliases(
+            event.OldResourceProperties.Aliases
+          );
+          var aliasesToDelete = [...prevAliases].filter(function (itm) {
+            return !aliases.has(itm);
+          });
+          var aliasesToAdd = [...aliases].filter(function (itm) {
+            return !prevAliases.has(itm);
+          });
+          if (aliasesToAdd.length + aliasesToDelete.length === 0) {
+            break;
+          }
+        }
         certificateArn = await requestCertificate(
           event.RequestId,
           props.AppName,
           props.EnvName,
-          props.DomainName,
-          props.SubjectAlternativeNames,
+          certDomain,
+          [...aliases],
```

Comment:
dumb question: why do we need to use the spread syntax here?

Reply:
no it's not dumb at all! I forgot to change this. it doesn't need to be a list.